run_unittests.py

```diff
@@ -2744,6 +2752,81 @@ def test_buildtype_setting(self):
         self.assertEqual(opts['debug'], True)
         self.assertEqual(opts['optimization'], '0')
 
+    def test_static_and_shared_library_usability(self):
+        '''
+        Test that static and shared libraries with various kinds of static
+        library internal dependencies are usable after installation, and that
+        the pkg-config files generated for such libraries have the correct
+        Libs: and Libs.private: lines.
+        '''
+        env = get_fake_env('', self.builddir, self.prefix)
+        cc = env.detect_c_compiler(False)
+        if cc.get_id() == 'msvc':
+            static_args = '-DPROVIDER_STATIC'
+            # FIXME: Can't reliably test mixed shared/static because of
+            # __declspec linkage issues and because it will greatly complicate
+            # the build files. Waiting for static_c_args support.
+            libtypes = ('static',)
+        else:
+            static_args = ''
+            libtypes = ('static', 'shared', 'both')
+        # Test
+        for libtype in libtypes:
+            oldprefix = self.prefix
+            # Install external library so we can find it
+            testdir = os.path.join(self.unit_test_dir, '35 both library usability', 'provider')
+            # install into installdir without using DESTDIR
+            installdir = self.installdir
```

Comment:
That's a bit hackish, you could test -uninstalled.pc files here, when that PR gets in. But I guess that's fine for a unit test.